go/vt/topo/tablet.go

```diff
@@ -263,15 +268,30 @@ func (ts *Server) GetTabletsByCell(ctx context.Context, cellAlias string, opt *G
 		return nil, err
 	}
 
-	tablets := make([]*TabletInfo, len(listResults))
+	var capHint int
+	if opt != nil && opt.KeyspaceShard == nil {
+		capHint = len(listResults)
+	}
+
+	tablets := make([]*TabletInfo, 0, capHint)
 	for n := range listResults {
 		tablet := &topodatapb.Tablet{}
 		if err := tablet.UnmarshalVT(listResults[n].Value); err != nil {
 			return nil, err
 		}
-		tablets[n] = &TabletInfo{Tablet: tablet, version: listResults[n].Version}
+		if opt != nil && opt.KeyspaceShard != nil && opt.KeyspaceShard.Keyspace != "" {
+			if opt.KeyspaceShard.Keyspace != tablet.Keyspace {
+				continue
+			}
+			if opt.KeyspaceShard.Shard != "" && opt.KeyspaceShard.Shard != tablet.Shard {
+				continue
+			}
+		}
+		tablets = append(tablets, &TabletInfo{Tablet: tablet, version: listResults[n].Version})
 	}
-
+	slices.SortFunc(tablets, func(i, j *TabletInfo) int {
```

Comment:
@mattlord I've converted the test to check exact tablets as suggested. A consequence of this is we need to sort the output

I can make this conditional for testing but I see we sort results in other places that do concurrent gets, so I'll hold off for now

Reply:
I suppose the sort could move to `tablet_test.go` if there is a concern about cost

Reply:
Updated: moved it to `tablet_test.go` until there's a need for a sort in the normal code